cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -320,13 +343,55 @@ private void updateKafkaConnectClusters(List<ConfigMap> update)   {
         }
     }
 
-    private void deleteConnectConnectClusters(List<Deployment> delete)   {
+    private void deleteKafkaConnectClusters(List<Deployment> delete)   {
         for (Deployment dep : delete) {
             log.info("Reconciliation: Kafka Connect cluster {} should be deleted", dep.getMetadata().getName());
             deleteKafkaConnectCluster(dep);
         }
     }
 
+    private void reconcileKafkaConnectS2I() {
+        log.info("Reconciling Kafka Connect S2I clusters ...");
+
+        Map<String, String> kafkaLabels = new HashMap(labels);
+        kafkaLabels.put(ClusterController.STRIMZI_TYPE_LABEL, KafkaConnectS2ICluster.TYPE);
+
+        List<ConfigMap> cms = configMapOperations.list(namespace, kafkaLabels);
+        List<DeploymentConfig> deps = deploymentConfigOperations.list(namespace, kafkaLabels);
+
+        List<String> cmsNames = cms.stream().map(cm -> cm.getMetadata().getName()).collect(Collectors.toList());
```

Comment:
I do this in some of the test too. We should probably factor out a method for this, but in another PR would be fine.

Reply:
Also, since we only end up using `contains()` on these collections of names the should be `Set<String>` rather than `List<String>`

Reply:
I think this should be part of the ClusterController refactoring.